src/docs/resources/architectural-overview.md

```diff
@@ -0,0 +1,966 @@
+[`RenderImage`](https://api.flutter.dev/flutter/rendering/RenderImage-class.html)
+renders
+an image, and
+[`RenderTransform`](https://api.flutter.dev/flutter/rendering/RenderTransform-class.html)
+applies a transformation before painting its child.
+
+![Differences between the widgets hierarchy and the element and render
+trees](../../images/arch-overview/trees.svg){:width="100%"}
+
+Most Flutter widgets are rendered by an object that inherits from the
+`RenderBox` subclass, which represents a `RenderObject` of fixed size in a 2D
+Cartesian space. `RenderBox` provides the basis of a _box constraint model_,
+establishing a minimum and maximum width and height for each widget to be
+rendered.
+
+To perform layout, Flutter walks the render tree in a depth-first traversal and
+**passes down size constraints** from parent to child. In determining its size,
+the child _must_ respect the constraints given to it by its parent. Children
+respond by **passing up a size** to their parent object within the constraints
+the parent established.
+
+![Constraints go down, sizes go
+up](../../images/arch-overview/constraints-sizes.svg){:width="100%"}
+
+At the end of this single walk through the tree, every object has a defined size
+within its parent’s constraints and is ready to be painted by calling the
+[`paint()`](https://api.flutter.dev/flutter/rendering/RenderObject/paint.html)
+method.
+
+The box constraint model is very powerful as a way to layout objects in _O(n)_
+time:
+
+- Parents can dictate the size of a child object by setting maximum and minimum
+  constraints to the same value. For example, the topmost render object in a
+  phone app constrains its child to be the size of the screen. (Children can
+  choose how to use that space. For example, they might just center what they
+  want to render within the dictated constraints.)
+- A parent can dictate the child’s width but give the child flexibility over
+  height (or dictate height but offer flexible over width). A real-world example
+  is flow text, which might have to fit a horizontal constraint but vary
+  vertically depending on the quantity of text.
+
+This model works even when a child object needs to know how much space it has
+available to decide how it will render its content. By using a
+[`LayoutBuilder`](https://api.flutter.dev/flutter/widgets/LayoutBuilder-class.html)
+widget, the child object can examine the passed-down constraints and use those
+to determine how it will use them, for example:
+
+```dart
+  Widget build(BuildContext context) {
+    return LayoutBuilder(
+      builder: (context, constraints) {
+        if (constraints.maxWidth < 600) {
+          return OneColumnLayout();
+        } else {
+          return TwoColumnLayout();
+        }
+      },
+    );
+```
+
+More information about the constraint and layout system, along with worked
+examples, can be found in the [Understanding
+Constraints](https://flutter.dev/docs/development/ui/layout/constraints) topic.
```

Comment:
Yay! Glad you linked to it! But, once again:
```
[Understanding constraints](/docs/development/ui/layout/constraints) topic.
```
Also note that we use sentence case for titles, so it's "Understanding constraints" with a lower case 'c'.